lib/AST/ASTImporter.cpp

```diff
@@ -3048,9 +3047,12 @@ ExpectedDecl ASTNodeImporter::VisitFunctionDecl(FunctionDecl *D) {
   }
   // Try to find a function in our own ("to") context with the same name, same
   // type, and in the same context as the function we're importing.
-  else if (!LexicalDC->isFunctionOrMethod()) {
+  else {
+    bool IsCImplicit = LexicalDC->isFunctionOrMethod() && D->isImplicit();
```

Comment:
We could have a lambda for this, which we could reuse at line 3070: `if (IsCImplicit(D) && IsCImplicit(FoundFunction)) { ... }`
Note, probably we want to check the Lexical decl in the "from" context, and not in the "to" context, `LexicalDC` is in the "to" context while `D` is in the "from" ctx.

Reply:
Also, it would be nice to explain what do we mean under "C implicit" functions, perhaps with an example or with a reference to the paragraph in the C standard.